src/DeviceInput/InputDevices/webDeviceInputSystem.ts

```diff
@@ -156,12 +155,58 @@ export class WebDeviceInputSystem implements IDeviceInputSystem {
             this._handleKeyActions();
             this._handlePointerActions();
             this._handleGamepadActions();
+            this._eventsAttached = true;
 
             // Check for devices that are already connected but aren't registered. Currently, only checks for gamepads and mouse
             this._checkForConnectedDevices();
         }
     }
 
+    /**
+     * Remove events from active input element
+     */
+     public removeEvents(): void {
+        if (this._elementToAttachTo) {
+            // Blur Events
+            this._elementToAttachTo.removeEventListener("blur", this._keyboardBlurEvent);
+            this._elementToAttachTo.removeEventListener("blur", this._pointerBlurEvent);
+
+            // Keyboard Events
+            this._elementToAttachTo.removeEventListener("keydown", this._keyboardDownEvent);
+            this._elementToAttachTo.removeEventListener("keyup", this._keyboardUpEvent);
+
+            // Pointer Events
+            this._elementToAttachTo.removeEventListener(this._eventPrefix + "move", this._pointerMoveEvent);
+            this._elementToAttachTo.removeEventListener(this._eventPrefix + "down", this._pointerDownEvent);
+            this._elementToAttachTo.removeEventListener(this._eventPrefix + "up", this._pointerUpEvent);
+            this._elementToAttachTo.removeEventListener(this._wheelEventName, this._pointerWheelEvent);
+        }
+
+        if (this._pointerInputClearObserver) {
+            this._engine.onEndFrameObservable.remove(this._pointerInputClearObserver);
+        }
+
+        this._eventsAttached = false;
+    }
+
+    /**
+     * Dispose of all the eventlisteners
+     */
+    public dispose(): void {
+        // Observables
+        this.onDeviceConnected = () => { };
+        this.onDeviceDisconnected = () => { };
+        this.onInputChanged = () => { };
+
+        if (this._elementToAttachTo) {
+            this.removeEvents();
+
+            // Gamepad Events
+            window.removeEventListener("gamepadconnected", this._gamepadConnectedEvent);
+            window.removeEventListener("gamepaddisconnected", this._gamepadDisconnectedEvent);

```

Comment:
why is this not part of removeEvents?

Reply:
I assume this is an old carryover from all of the changes.  I moved this to unconfigure